examples/evm/call-contract-gas-estimation/CallContractGasEstimation.sol

```diff
@@ -0,0 +1,73 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import { AxelarExecutable } from '@axelar-network/axelar-gmp-sdk-solidity/contracts/executable/AxelarExecutable.sol';
+import { IAxelarGateway } from '@axelar-network/axelar-gmp-sdk-solidity/contracts/interfaces/IAxelarGateway.sol';
+import { IAxelarGasService } from '@axelar-network/axelar-gmp-sdk-solidity/contracts/interfaces/IAxelarGasService.sol';
+import { IERC20 } from '@axelar-network/axelar-gmp-sdk-solidity/contracts/interfaces/IERC20.sol';
+
+/**
+ * @title CallContractGasEstimation
+ * @notice Send a message from chain A to chain B and stores gmp message
+ */
+contract CallContractGasEstimation is AxelarExecutable {
+    string public message;
+    string public sourceChain;
+    string public sourceAddress;
+    IAxelarGasService public immutable gasService;
+    uint256 public constant GAS_LIMIT = 200000;
+
+    event Executed(string _from, string _message);
```

Comment:
add a getter to estimate gas on this contract for composability? It'll internally call `gasService.estimateGas` with the right gas limit set. Calling contracts and the example script can use that instead

Reply:
Implemented!